### DISCUSSION TITLE
Passing symbols from analyzer to code fix

### DISCUSSION THREAD

**Discussion by Aidiakapi** [dotnet/roslyn#53440] [Q&A]
I'm looking to pass a symbol from a Code Analyzer to a Code Fix. It seems like the only communication method is the property bag, which only works on strings. The solution then seems to be to serialize and deserialize the symbol reference.

You have [`compilation.GetTypeByMetadataName()`][1] for deserialization, which seems [to not be 100% reliable][2], and the [API to get the metadata name is internal][3] (unless that has been resolved at some point).

What's the recommended way to do this? It'd seem a fairly typical scenario for code fixes to get some information from the analyzers.

[1]: https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.compilation.gettypebymetadataname?view=roslyn-dotnet-3.9.0
[2]: https://github.com/dotnet/roslyn/issues/52037
[3]: https://github.com/dotnet/roslyn/issues/1891

**Comment by CyrusNajmabadi** [comment]
Have the analyzer include enough information so that the fixer can go re-retrieve the symbol from the compilation it is operating in.

**Reply by Aidiakapi** [reply]
Well, that's what I'm asking for, a way to serialize the information about the symbol that's necessary to re-retrieve it from the compilation.

If you were implying that I shouldn't directly pass the symbol, but instead perhaps re-do the analysis to come to the same symbols once again, I'd rather not do this, because the code fixes run per-diagnostic, while the analysis does some initial set-up to speed up the checks.
